ios/Classes/Models/HMSPIPAction.swift

```diff
@@ -0,0 +1,229 @@
+//
+//  HMSPIPAction.swift
+//  hmssdk_flutter
+//
+//  Created by Govind on 12/01/23.
+//
+
+import Foundation
+import HMSSDK
+import AVKit
+import SwiftUI
+
+@available(iOS 15.0, *)
+class HMSPIPAction {
+    static var pipVideoCallViewController: UIViewController?
+    static var pipController: AVPictureInPictureController?
+    static var model: PiPModel?
+    static func pipAction(_ call: FlutterMethodCall, _ result: @escaping FlutterResult, _ hmsSDK: HMSSDK?, _ swiftHmssdkFlutterPlugin: SwiftHmssdkFlutterPlugin) {
+        switch call.method {
+        case "setup_pip":
+            setupPIP(call, result, hmsSDK, swiftHmssdkFlutterPlugin)
+
+        case "start_pip":
+            startPIP()
+
+        case "stop_pip":
+            stopPIP()
+            
+        case "is_pip_available":
+            isPIPAvailable(result)
+        
+        case "is_pip_active":
+            isPIPActive(result)
+            
+        case "change_track_pip":
+            changeTrack(call, result, hmsSDK)
+        
+        case "change_text_pip":
+            changeText(call, result, hmsSDK)
+            
+        case "destroy_pip":
+            disposePIP(result)
+
+        default:
+            result(FlutterMethodNotImplemented)
+        }
+    }
+    
+    static func setupPIP(_ call: FlutterMethodCall, _ result: @escaping FlutterResult, _ hmsSDK: HMSSDK?,_ swiftHmssdkFlutterPlugin: SwiftHmssdkFlutterPlugin) {
+        
+        guard AVPictureInPictureController.isPictureInPictureSupported() else {
+            result(HMSErrorExtension.getError("\(#function) PIP is not supported"))
+            return }
+        
+        guard let uiView = UIApplication.shared.keyWindow?.rootViewController?.view else {
+            result(HMSErrorExtension.getError("\(#function) Failed to setup PIP"))
+            return }
+            
+        let pipVideoCallViewController = AVPictureInPictureVideoCallViewController()
+        
+        self.pipVideoCallViewController = pipVideoCallViewController
```

Comment:
Since it's a property of the class, ensure that `pipVideoCallViewController` reset to nil when leave, onRemovedFromRoom, destroy of PIP Controller, etc functions are invoked